packages/cosmos_ui_components/lib/components/cosmos_switch.dart

```diff
@@ -1,101 +1,59 @@
 import 'package:cosmos_ui_components/cosmos_theme.dart';
-import 'package:cosmos_ui_components/utils/global_constants.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 
-class CosmosSwitch extends StatefulWidget {
+class CosmosSwitch extends StatelessWidget {
   const CosmosSwitch({
     required this.checked,
+    required this.toggleAsset,
+    required this.checkedGradientStops,
+    required this.gradientColors,
     this.onChanged,
     Key? key,
   }) : super(key: key);
 
   final bool checked;
   final ValueChanged<bool>? onChanged;
-
-  @override
-  CosmosSwitchState createState() => CosmosSwitchState();
-
-  @override
-  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
-    super.debugFillProperties(properties);
-    properties
-      ..add(ObjectFlagProperty<ValueChanged<bool>?>.has('onChanged', onChanged))
-      ..add(DiagnosticsProperty<bool>('checked', checked));
-  }
-}
-
-class CosmosSwitchState extends State<CosmosSwitch> with SingleTickerProviderStateMixin {
-  late Animation<Alignment> _circleAnimation;
-  late AnimationController _animationController;
-
-  @override
-  void initState() {
-    super.initState();
-    _animationController = AnimationController(vsync: this, duration: const Duration(milliseconds: 200));
-    _circleAnimation = AlignmentTween(
-      begin: widget.checked ? Alignment.centerRight : Alignment.centerLeft,
-      end: widget.checked ? Alignment.centerLeft : Alignment.centerRight,
-    ).animate(
-      CurvedAnimation(
-        parent: _animationController,
-        curve: Curves.linear,
-      ),
-    );
-  }
+  final String toggleAsset;
```

Comment:
you've now removed the `toggle.png` which makes the asset `required` and will make the starport template and emeris app not compiling. the param should be optional and have default value of `toggle.png` that should be embedded into the library